### ISSUE TITLE
current `vpn-slice` fails with `(22, 'Invalid argument')`

### ISSUE THREAD

**Issue by cresh** [dlenski/vpn-slice#124]
Using `vpn-slice` from current `master` in a Docker container on macOS fails with "Invalid argument":

```
# vpn-slice 10.0.0.0./8 example.com
(22, 'Invalid argument')
```

It even fails in the same way without any arguments:

```
# vpn-slice
(22, 'Invalid argument')
```

Using revision [fde18b767df20bcea842084f2e03eb0723f9b4ea](https://github.com/dlenski/vpn-slice/archive/fde18b767df20bcea842084f2e03eb0723f9b4ea.zip) still works as expected.

You can find a complete `strace` here: [strace-vpn-slice.log](https://github.com/dlenski/vpn-slice/files/10055970/strace-vpn-slice.log)

**Comment by vadim-cherepanov** [comment]
Any news on this one? I exactly have the same problem.

**Comment by br3t3s** [comment]
Hi,

I'm facing the same issues with fresh instalation ubuntu server 20.04 and 22.04. Can someone investigate what is wrong? Even as a root user execute vpn-slice is impossible. Always get (22, 'Invalid argument')

**Comment by vadim-cherepanov** [comment]
From what I learned, there was a series of recent merges after the commit mentioned in the OP, some of which caused the problem.

```
M = the current master with the OP bug
^
|\
|  | some external PRs, now merged in
|/
F = the latest commit `fde18b767...` before the merges, no OP bug
^
|
A = the latest official release v0.16.1
```
Now, there was also a recent update to `python-setuptools` (PST), which introduced another bug in versions 65.6.1-2 and maybe some earlier. This bug was resolved in the PST 65.6.3. Now, A is affected by the PST bug, and cannot be compiled with the PST 65.6.1-2 and maybe some earlier versions. While, somehow, F and M are not affected.

Overall, to avoid the OP bug go back to A or F. Try the latest official release A, and if does not compile due to the PST bug, either update the PST or move forward to F but not later.

**Comment by parantapa** [comment]
I was also facing this problem.

Atleast in my case the following line is the issue:
https://github.com/dlenski/vpn-slice/blob/9ecb50db80fe912f99bc0996b4465aeb97f65ac2/vpn_slice/linux.py#L145

When /etc/resolv.conf is not a link, this raises OSError.
Catching the OSError and returning False fixes the issue for me.

**Comment by DimitriPapadopoulos** [comment]
Does  #126 fix this issue for all of you?

**Comment by dlenski** [comment]
See 
https://github.com/dlenski/vpn-slice/pull/118#issuecomment-1624177196 for why this happened.

Fixed by reverting #118 in a640258.

PRs to improve the error reporting would be quite welcome :-)
